### PR TITLE
Create an Events table and add the pages

### LINKED ISSUE #9: Add an Event Page
Events are defined by the following characteristics:

name | type
--- | ---
name | string
start | datetime
end | datetime
description | string | use [Kramdown](https://github.com/gettalong/kramdown) to parse

### PR DESCRIPTION
**Pull request by ChaelCodes**
# Description
Add a table to hold Events. Events are events that profiles can attend.
closes #9 

- Scaffold Events
- Customize scaffold
   * remove notice from individual pages
   - change datetime_field to datetime_local_field
   - specs use factories
   - specs use subjects
   - refactor authenticate_user
   - fix bugs in user/profile policy revealed by refactor
   - customize factory
   - move root to events
- Fixes to Scaffold for Rubocop

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    @event.destroy
+    respond_to do |format|
+      format.html { redirect_to events_url, notice: 'Event was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+
+  def create_event
```
Comment (PinkFlufflyLlama) [inline app/controllers/events_controller.rb:62]: Do I need a comment?

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class Event < ApplicationRecord
```
Comment (PinkFlufflyLlama) [inline app/models/event.rb:3]: Doesn't this class need a top level comment too?

File: db/schema.rb [hunk 3]
```diff
@@ -10,11 +10,20 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2021_07_04_151452) do
+ActiveRecord::Schema.define(version: 2021_07_11_152754) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "events", force: :cascade do |t|
```
Comment (PinkFlufflyLlama) [inline db/schema.rb:18]: does this need to be in a migration?

Comment (ChaelCodes) [reply db/schema.rb:18]: No, the schema is where the final database structure lives. There is a create table migration.

File: app/models/event.rb (hunk 2 above)
Comment (ChaelCodes) [reply app/models/event.rb:3]: Yeah! It's super weird Rubocop didn't complain.

**Pull request merged**